### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by adeelq**

### CODE REVIEW

**Comment by cdiacon** [comment]
@adeelq Upgrade script should match the 2.0.6 version which is the latest at the moment, on 2.0.4 installations the upgrade changes will not run.

**Comment by cdiacon** [comment]
@adeelq also for performance reasons the new column "send_staus" for email_campaign needs to have a database index as it's major filter for this table.

File: Setup/UpgradeSchema.php [hunk 1]
```diff
@@ -22,7 +22,7 @@ public function upgrade(SchemaSetupInterface $setup, ModuleContextInterface $con
             //remove quote table
             $connection->dropTable($setup->getTable('email_quote'));
         }
-        if (version_compare($context->getVersion(), '2.0.4') < 0) {
+        if (version_compare($context->getVersion(), '2.0.6') < 0) {
```
Comment (cdiacon) [inline Setup/UpgradeSchema.php]: @adeelq version compare needs : version1, version2 and operator as params  to keep it with the code style : 

```
if (version_compare($context->getVersion(), '2.0.6', '<')) {
```

File: Model/ResourceModel/Campaign.php [hunk 2]
```diff
@@ -15,22 +15,29 @@ public function _construct()
...
+     * @param $ids
      * @param $message
+     * @param $sendId
      * @throws \Magento\Framework\Exception\LocalizedException
      */
-    public function setMessage($campaignId, $message)
+    public function setMessage($ids, $message, $sendId = false)
     {
         try {
+            $ids = implode("', '", $ids);
```
Comment (cdiacon) [inline Model/ResourceModel/Campaign.php]: @adeelq remove the space after the comma, we don't need it

File: Model/Sync/Campaign.php [hunk 3]
```diff
@@ -160,36 +165,71 @@ public function sendCampaigns()
...
+     * @param $sendStatus
+     * @param $sendIdCheck
      * @return mixed
      */
-    protected function _getEmailCampaigns($storeIds)
+    protected function _getEmailCampaigns($storeIds, $sendStatus = 0, $sendIdCheck = false)
     {
         $emailCollection = $this->_campaignCollection->create()
-            ->addFieldToFilter('is_sent', ['null' => true])
+            ->addFieldToFilter('send_status', ['eq' => $sendStatus])
```
Comment (cdiacon) [inline Model/Sync/Campaign.php]: @adeelq not sure if we need the ['eq' operator here, you can specify only the value here

Comment (adeelq) [inline Model/Sync/Campaign.php]: @cdiacon sure you can just like in m1...see example https://mage2.pro/t/topic/1249

Comment (cdiacon) [inline Model/Sync/Campaign.php]: you need/can do this if you need crazy equation like this: 

```
    $this->addFieldToFilter('severity', ['eq' => $severity * 1]);
```

**Pull request merged**